logFilter.js

```diff
@@ -17,11 +17,11 @@ function filterLogsOnDate(logs, dateFilter) {
   logs.forEach((log) => {
     const timestampDate = new Date(log.timestamp)
 
-    if (parsedDateFilter.toDateString() === timestampDate.toDateString()) {
+    if (parsedDateFilter.getUTCDate() === timestampDate.getUTCDate()) {
```

Comment:
This is the right idea but you're missing some cases. `getUTCDate` returns the day of the month for a Date object without doing any timezone conversion. The issue here is that if I ask for logs for `2019-09-30` I will get logs for any 30th day such as `2019-08-30` because you're only checking that the `30` matches, not the entire date.